driver-core/src/main/com/mongodb/client/model/expressions/DocumentExpression.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.client.model.expressions;
+
+import org.bson.conversions.Bson;
+import org.bson.types.Decimal128;
+
+import java.time.Instant;
+
+import static com.mongodb.client.model.expressions.Expressions.of;
+
+/**
+ * Expresses a document value. A document is an ordered set of fields, where the
+ * key is a string value, mapping to a value of any other expression type.
+ */
+public interface DocumentExpression extends Expression {
```

Comment:
The BSON spec calls an ordered collection of key/value pairs (with no key duplicates) a "document", while the MongoDB documentation calls it an "object". The "expressions API" splits the document/object type into two: "MQL document" and "MQL map<T>" in such a way that:

1. Any "MQL document" and any "MQL map" is an "BSON document/object".
2. Not any "BSON document/object" is "MQL document".
3. Not any "BSON document/object" is "MQL map<T>", unless `T` is `Expression` (which is supposed to be named "MQL value").

Due to item 2, the meaning of the word "document" in "BSON document/object" is different from that in "MQL document", i.e., we have overloaded its meaning. We should use a different word instead of overloading the existing one. Even "MQL record" appears to be better than "MQL document", because despite being used in "Java records", it is not used within the BSON/MongoDB domain. It is much more likely (almost guaranteed) that "MQL document" will be confused with "BSON document/object", and much less likely that "MQL record" will be confused with "Java record".